packages/@aws-cdk/aws-apigateway/lib/deployment.ts

```diff
@@ -0,0 +1,160 @@
+import crypto = require('crypto');
+import { cloudformation, DeploymentId } from './apigateway.generated';
+import { RestApiRef } from './restapi-ref';
+
+export interface DeploymentProps  {
+    /**
+     * The Rest API to deploy.
+     */
+    api: RestApiRef;
+
+    /**
+     * A description of the purpose of the API Gateway deployment.
+     */
+    description?: string;
+
+    /**
+     * When an API Gateway model is updated, a new deployment will automatically be created.
+     * If this is true (default), the old API Gateway Deployment resource will not be deleted.
+     * This will allow manually reverting back to a previous deployment in case for example
+     *
+     * @default true
+     */
+    retainDeployments?: boolean;
+}
+
+/**
+ * A Deployment of a REST API.
+ *
+ * An immutable representation of a RestApi resource that can be called by users
+ * using Stages. A deployment must be associated with a Stage for it to be
+ * callable over the Internet.
+ *
+ * Normally, you don't need to define deployments manually. The RestApi
+ * construct manages a Deployment resource that represents the latest model. It
+ * can be accessed through `restApi.latestDeployment` (unless `deploy: false` is
+ * set when defining the `RestApi`).
+ *
+ * If you manually define this resource, you will need to know that since
+ * deployments are immutable, as long as the resource's logical ID doesn't
+ * change, the deployment will represent the snapshot in time in which the
+ * resource was created. This means that if you modify the RestApi model (i.e.
+ * add methods or resources), these changes will not be reflected unless a new
+ * deployment resource is created.
+ *
+ * To achieve this behavior, the method `addToLogicalId(data)` can be used to
+ * augment the logical ID generated for the deployment resource such that it
+ * will include arbitrary data. This is done automatically for the
+ * `restApi.latestDeployment` deployment.
+ *
+ * Furthermore, since a deployment does not reference any of the REST API
+ * resources and methods, CloudFormation will likely provision it before these
+ * resources are created, which means that it will represent a "half-baked"
+ * model. Use the `addDependency(dep)` method to circumvent that. This is done
+ * automatically for the `restApi.latestDeployment` deployment.
+ */
+export class Deployment extends cdk.Construct {
+    public readonly deploymentId: DeploymentId;
+    public readonly api: RestApiRef;
+
+    private readonly resource: LatestDeploymentResource;
+
+    constructor(parent: cdk.Construct, id: string, props: DeploymentProps) {
+        super(parent, id);
+
+        this.resource = new LatestDeploymentResource(this, 'Resource', {
+            description: props.description,
+            restApiId: props.api.restApiId,
+        });
+
+        if (props.retainDeployments === undefined || props.retainDeployments) {
+            this.resource.options.deletionPolicy = cdk.DeletionPolicy.Retain;
```

Comment:
This is the default behavior when using the console, so I thought this might make sense as a default, but I think people don't really care about this, so maybe I'll change the default.